### PR TITLE
Add a CLI to enable the usage of the codegen offline

### PR DESCRIPTION
**Pull request by andreaTP**
A proposed technical solution for: https://github.com/Apicurio/apicurio-registry/issues/2656

This is merely hacked together for now, but, a possible workflow would be:
- generate in CI the source code from this CLI
- copy all the files from the output `zip` archive to the relevant `/src/main` folder
- check that nothing changed using `git`

### CODE REVIEW

**Comment by andreaTP** [comment]
Example published CLIs:
https://github.com/andreaTP/apicurio-codegen/releases/tag/0.0.1-SNAPSHOT

**Comment by andreaTP** [comment]
ready for final review, not automated yet, but incorporated a generated configuration for GraalVM reflection, and the tool looks to be working as expected.

File: cli/tools/extractRegisterForReflection.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+//DEPS org.reflections:reflections:0.10.2
+//DEPS io.apicurio:apicurio-data-models:1.1.26
+
+import java.util.*;
+import org.reflections.*;
+
+import static java.lang.System.*;
+import static org.reflections.scanners.Scanners.*;
+
+public class extractRegisterForReflection {
```
Comment (EricWittmann) [inline cli/tools/extractRegisterForReflection.java:11]: Why the lower case `e` in this class name?  That's pretty unusual in Java so I'm assuming it was intentional?

File: cli/tools/extractRegisterForReflection.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+    public static void main(String... args) {
+        Reflections reflections = new Reflections("io.apicurio");
+
+        Set<Class<?>> subTypes = reflections.get(SubTypes.of(io.apicurio.datamodels.core.models.Node.class).asClass());
```
Comment (EricWittmann) [inline cli/tools/extractRegisterForReflection.java:16]: Nice.

**Review by EricWittmann — APPROVED** [review]

File: cli/tools/extractRegisterForReflection.java (hunk 1 above)
Comment (andreaTP) [reply cli/tools/extractRegisterForReflection.java:11]: Due to Java limitations, the name of the class should be strictly equal to the file name and I usually name scripts starting with a lowercase.

Given the fact that `jbang` is a pretty recent technology I doubt that there is any "strong" convention, in the front page they use lower-case scripts names though: https://www.jbang.dev/

Anyhow, not fussy about it, just suggest a preferred name and I'll fix it as you prefer 🙂

**Pull request merged**